src/tsdb.c

```diff
@@ -15,52 +15,95 @@
 
 #include <inttypes.h>
 #include <math.h>
+#include <stdlib.h>
 #include "rmutil/alloc.h"
 #include "rmutil/logging.h"
 #include "rmutil/strings.h"
 
-static Series *lastDeletedSeries = NULL;
 static RedisModuleString *renameFromKey = NULL;
 
+static void deleteReferenceToDeletedSeries(Series *series,
+                                           SERIES_RELATION relation,
+                                           RedisModuleString *keyName) {
+    switch (relation) {
```

Comment:
The logic here should be simple:
1. we should only get the Series and the keyname
2. for each rule:
     2.1 open dest key, if doesn't exists delete rule
     2.2 if the destKey->srcKey doesn't equal our keyName, delete rule
     2.3 if the destKey->srcKey is NULL, delete rule
3. if series->srcKey is not NULL:
     3.1 open the key, if it doesn't exists set srcKey=NULL
     3,2 if it exists but there's no matching rule, set srckey=NULL